pkg/envoy/admin/client.go

```diff
@@ -82,6 +82,10 @@ func (a *envoyAdminClient) getOrCreateSigningKey(mesh string) (string, error) {
 
 func (a *envoyAdminClient) adminAddress(dataplane *mesh_core.DataplaneResource) string {
 	ip := dataplane.GetIP()
+	// TODO: this will work perfectly fine with K8s, but will fail for Universal
+	// The real allocated admin port is part of the DP metadata, but it is attached to a particular CP (the one that
+	// bootstrapped the DP), so we can not reliably use that. A better approach would be to include the admin port
+	// in the DataplenInsights.
```

Comment:
`DataplenInsights` typo.

It's not necessary a better approach. Maybe it makes more sense to encode this in Dataplane itself.

> the one that bootstrapped the DP

not exactly. DP metadata is in the CP that DP is connected to. It could get bootstrap config from instance 1, but then connect via xds to instance 2